File: Haskell/1343B.hs
{-# LANGUAGE LambdaCase #-}

main :: IO ()
main = interact $ unlines . map (f . solve . read) . tail . words

f :: Maybe [Int] -> String
f = \case
  Just xs -> "YES\n" <> unwords (map show xs)
  Nothing -> "NO"

solve :: Int -> Maybe [Int]
solve n
  | n `mod` 4 /= 0 =  Nothing
  | otherwise = Just $ [2, 4..n] <> [1, 3.. n - 2] <> [(n - 1) + n `div` 2]

File: CPP/1343B.cpp
#include <iostream>

using namespace std;

int main()
{
  int t, n;

  for(cin >> t; t--;)
  {
    cin >> n;
    n /= 2;
    if (n & 1) 
    {
      cout << "NO" << endl;
      continue;
    }

    cout << "YES" << endl;
    for(int i = 1; i <= n; ++i) cout << i * 2 << " ";
    for(int i = 1; i <  n; ++i) cout << i * 2 - 1 << " "; 

    cout << (n * 3) - 1 << endl;
  }
}